.github/workflows/build-commit.yml

```diff
@@ -0,0 +1,48 @@
+name: Build
+on: [push, pull_request]
+jobs:
+  build_amd64:
+    name: Build (AMD64 - CentOS 7 Target)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Build code-server Binary
+        run: yarn && bash ./scripts/ci.bash  
```

Comment:
We don't need to run yarn for any of this right? I think the scripts should execute properly without needing to run yarn since we download the cache which includes `node_modules`.

Reply:
It's still a good idea to run `yarn` in case dependencies changed.

Reply:
We don't copy the `node_modules` into the Docker container so these would just go unused at least for the Linux builds. We could copy them in but the build script runs `yarn` anyway in the code-server directory so I don't think it's necessary.

And actually I misspoke a little, the cache only includes VS Code dependencies, not code-server. The caching situation could really use improvement...

Reply:
I think ideally we'd run yarn here with some sort of caching and copy it into the Docker container when building.

Reply:
Except we have to run yarn inside the Docker container to get the native modules built correctly.

Reply:
I'd strongly recommend circleci over this GCP caching you guys are doing. It'll be much faster since the cache will be colocated with the circle ci infrastructure instead of fetching from gcp.

@sr229

Reply:
Yeah honestly that sounds like it'd be much nicer.

Reply:
Example for a Go library (but should apply similarly to code-server) at https://github.com/nhooyr/websocket/blob/27104bdeb98d981a335da9faeb0efd1b8d4f8b65/.circleci/config.yml

https://github.com/nhooyr/websocket/runs/215657031